vector/src/main/java/im/vector/app/features/matrixto/MatrixToSource.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.matrixto
+
+import im.vector.app.features.analytics.plan.JoinedRoom
+import im.vector.app.features.analytics.plan.ViewRoom
+import java.lang.IllegalArgumentException
+
+enum class MatrixToSource {
```

Comment:
`MatrixToSource` may be a confusing name. This is not `Matrix to source`, but `MatrixTo source`, AFAIU.

Maybe rename to `SourceOfMatrixTo`, or `OriginOfMatrixTo` to avoid this confusion?